mne_bids/write.py

```diff
@@ -1317,29 +1344,44 @@ def write_anat(bids_root, subject, t1w, session=None, acquisition=None,
                                       suffix='T1w.nii.gz')
 
     # Check if we have necessary conditions for writing a sidecar JSON
-    if trans is not None:
-
-        # get trans and ensure it is from head to MRI
-        trans, _ = _get_trans(trans, fro='head', to='mri')
-
-        if not isinstance(raw, BaseRaw):
-            raise ValueError('`raw` must be specified if `trans` is not None')
-
-        # Prepare to write the sidecar JSON
-        # extract MEG landmarks
-        coords_dict = _extract_landmarks(raw.info['dig'])
-        meg_landmarks = np.asarray((coords_dict['LPA'],
-                                    coords_dict['NAS'],
-                                    coords_dict['RPA']))
-
-        # Transform MEG landmarks into MRI space, adjust units by * 1e3
-        mri_landmarks = apply_trans(trans, meg_landmarks, move=True) * 1e3
-
-        # Get landmarks in voxel space, using the mgh version of our T1 data
+    if trans is not None or landmarks is not None:
         t1_mgh = nib.MGHImage(t1w.dataobj, t1w.affine)
-        vox2ras_tkr = t1_mgh.header.get_vox2ras_tkr()
-        ras2vox_tkr = np.linalg.inv(vox2ras_tkr)
-        mri_landmarks = apply_trans(ras2vox_tkr, mri_landmarks)  # in vox
+
+        if landmarks is not None:
+            if raw is not None and verbose:
+                warn('Using provided `landmarks` instead of using ' +
+                     'the `trans` to convert the landmarks from `raw` ' +
+                     'space')
```

Comment:
you don't use raw in this code black. I am confused.

Reply:
In this case, if the user passed the raw for the dig coordinates, I thought they would think that the coordinates would be used and so would warn if not.